lib/api/core/plugins/pluginsManager.js

```diff
@@ -876,20 +884,33 @@ function triggerPipes(pipes, event, data) {
 }
 
 /**
- * For a specific event, return the corresponding wildcard
+ * For a specific event, return the correspondings wildcards
  * @example
- *  getWildcardEvent('data:create') // return 'data:*'
- * @param {string} event
- * @returns {String} wildcard event
+ *  getWildcardEvents('data:create') // return ['data:*']
+ *  getWildcardEvents('data:beforeCreate') // return ['data:*', 'data:before*']
+ * @param {String} event
+ * @returns {Array<String>} wildcard events
  */
-function getWildcardEvent (event) {
-  const indexDelimiter = event.indexOf(':');
-  if (indexDelimiter !== 1) {
-    return event.substring(0, indexDelimiter+1) + '*';
+const getWildcardEvents = _.memoize(event => {
+  const delimIndex = event.lastIndexOf(':');
+  
+  if (delimIndex === -1) {
+    return null;
```

Comment:
to prevent breaking code using the return value of this function without first checking that it didn't return null:

```suggestion
    return [];
```